janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphBlueprintsGraph.java

```diff
@@ -287,6 +289,18 @@ public GraphTransaction() {
             super(JanusGraphBlueprintsGraph.this);
         }
 
+        @Override
+        public void commit() {
+            super.commit();
+            doClose();
```

Comment:
This should be placed in a try - finally block to ensure threadlocals are cleaned even if commit/rollback fails.